jackson/src/main/resources/reference.conf

```diff
@@ -38,18 +38,18 @@ akka.serialization.jackson {
 #//#jackson-modules
 
 akka.serialization.jackson {
-  jackson-json {
-    # No compression for message class that implements the Jsonable marker interface.
-    # FIXME this config will probably change in Akka 2.6.0-M4
-    compress-larger-than = off
-  }
+  # compression is disabled for jackson-json binding in play/reference-overrides.conf
 
   jackson-json-gzip {
-    # The serializer will compress the payload if the message class
-    # implements the CompressedJsonable marker interface and the payload
-    # is larger than this value. Only used for remote messages and persistent data
-    # within the cluster of the service.
-    compress-larger-than = 32 KiB
+    compression {
+      algorithm = gzip
```

Comment:
I'm confused by duplicate semantic (`gzip`) on the path of this setting. If we will use another algorithm, the path will contain `gzip` and `lz4`.
```
akka.serialization.jackson.jackson-json-gzip.compression.algorithm = lz4
```
Are we sure that it's will be understandable for users?

Reply:
Good point. 

@patriknw, I don't see any of this in Akka. Is `akka.serialization.jackson.jackson-json-gzip` Lagom only thing? And if so, shouldn't we move it to Akka? 

And of course, remove `-gzip` at the end.

Reply:
The `CompressedJsonable` marker interface is bound to the configuration in `akka.serialization.jackson.jackson-json-gzip`.

If lz4 is later supported and users want to switch to that they will bind CompressedJsonable to a different config section `akka.serialization.jackson.jackson-json-lz4`.

```
akka.serialization.jackson.jackson-json-lz4 {
  compression {
    algorithm = lz4
  }
}
```

That binding would look like:
```
akka.actor {
  serializers {
    jackson-json-lz4 = "akka.serialization.jackson.JacksonJsonSerializer"
  }
  serialization-bindings {
    "com.lightbend.lagom.serialization.CompressedJsonable" = jackson-json-lz4
  }
  serialization-identifiers {
    "jackson-json-lz4" = 1000008
  }
}
```

At that point Lagom might choose to provide a different marker interface so that users don't have to worry about this config. `CompressedLz4Jsonable`.

At that point Lagom might also consider deprecating `CompressedJsonable` in favor of introducing `CompressedGzipJsonable` for consistent naming.

Reply:
in that case, why the duplication? 

Does it make sense to have the following?

```
akka.serialization.jackson.jackson-json-gzip {
  compression {
    algorithm = some-other-algo
  }
}
```

Reply:
@patriknw If I right understand the **key** of the setting should contain a semantic of his body because it used as a **value** in another setting? For example:
```YAML
akka.serialization.jackson.jackson-json-lz4-64kb {
  compression {
    algorithm = lz4
    compress-larger-than = 64 KiB
  }
}

# Here I need to know what a serializer is used by its name.
# Otherwise I need to go to settings of serializer for know what it doing
foo.bar = jackson-json-lz4-64kb 
```

Reply:
Maybe we should just rename `jackson-json-gzip` to `jackson-json-compressed`. By default with `algorithm = gzip`, but later if lz4 is supported users can only change
```
akka.serialization.jackson.jackson-json-compressed.algorithm = lz4
```

Reply:
> Maybe we should just rename `jackson-json-gzip` to `jackson-json-compressed`. By default with `algorithm = gzip`, but later if lz4 is supported users can only change

Will Jackson (or `JacksonJsonSerializer`) be able to know what algorithm was used when reading back a persisted, compressed event? Is that setup added on the manifest or can the de/ser code infer the compression alg some other way?

If `yes`, then I agree that renaming `jackson-json-gzip` to `jackson-json-compressed` is preferable.

Reply:
For gzip it checks for the magic. It wasn't completely clear how to include the magic for lz4, but something similar should be possible. Especially since we can change the `compress-larger-than` it's important to be able to detect if it's compressed or not.

I'll change to `jackson-json-compressed`. That is better.

Reply:
done